### ISSUE TITLE
Build all template package versions in servicing for source-build

### ISSUE THREAD

**Issue by MichaelSimons** [dotnet/templating#4337]
Currently building release/6.01xx doesn't produce the template packages.  This is because `IsPackable` is set to false for preview and servicing releases as described [here](https://github.com/dotnet/templating/blob/release/6.0.1xx/template_feed/Microsoft.DotNet.Common.ProjectTemplates.2.1/Microsoft.DotNet.Common.ProjectTemplates.2.1.csproj#L9).  This doesn't work well for .NET source-build which needs to product the entire product for every build (e.g. previews, rtm, servicing, etc).

Would it be amenable to conditionally build all template packages during source-build (e.g. Condition="'$(DotNetBuildFromSource)' == 'true'")?

Currently source-build treats these packages as text-only packages that are checked in/reverse engineered.  This solution is not ideal as it requires manual updates, is a maintenance burden, and in general is problematic. 

This issue is similar in nature to https://github.com/dotnet/aspnetcore/issues/39471.

**Comment by vlada-shubina** [comment]
@MichaelSimons do you need to build all the template packages? (even for the old version like 1.0, 2.0 etc).

We decided to build the only respective package version in the branch:
- release/5.0.xx - builds only `Microsoft.DotNet.Common.ProjectTemplates.5.0`
- release/6.0.xx - builds only `Microsoft.DotNet.Common.ProjectTemplates.6.0`
- `main`  - to be `release/7.0.xx` - builds only `Microsoft.DotNet.Common.ProjectTemplates.7.0`

Also we agreed not to publish package for servicing unless there is a need to, that's why it is disabled. 
I can enable publishing for source build only as you recommended above; however please confirm if you need all the packages or only 6.0. Starting 7.0 only 7.0 package exist anyway,

**Comment by MichaelSimons** [comment]
@vlada-shubina, The requirement is that everything required for a version of build, must be built from source.  For example, if the 6.0 build references multiple versions of the templates, then we would need each of those versions included as part of the source-build.  It is my understanding that the 6.0 installer build references 6.0, 5.0, 3.1, 3.0, and 2.1 templates as seen here - https://github.com/dotnet/installer/blob/release/6.0.1xx/src/redist/targets/BundledTemplates.targets#L88.  We either need to build all these versions as part of source build, or if these bundled templates are not necessary for source-build, then an alternative option would be to conditionally disable these from source-build.  Do the bundled templates ship with the SDK?  I only see the 6.0 templates with a 6.0 SDK.  If that is the case, maybe these should be excluded from source-build and then we would only need to ensure the 6.0 templates are built with the 6.0 source-build.

**Comment by MichaelSimons** [comment]
cc @dotnet/source-build-internal

**Comment by vlada-shubina** [comment]
@MichaelSimons 

unfortunately enabling shipping for `Microsoft.DotNet.Common.ProjectTemplates.5.0`, `Microsoft.DotNet.Common.ProjectTemplates.3.1` and `Microsoft.DotNet.Common.ProjectTemplates.2.1` won't likely help in this case.

the versions included to installer are listed here:
https://github.com/dotnet/installer/blob/701b7698d699e5c414bb8e49a52f426f36134de2/eng/Versions.props#L110-L148

As you can see, `Microsoft.DotNet.Common.ProjectTemplates.5.0` version is 5.0.13, and it is built from `release/5.0` branch;`Microsoft.DotNet.Common.ProjectTemplates.3.1` version is 3.1.15, and it is built from `release/3.1` branch.

From `release/6.0.1xx` branch, only  `Microsoft.DotNet.Common.ProjectTemplates.6.0` and `Microsoft.DotNet.Common.ItemTemplates` will be bundled to installer (6.0.1xx). Earlier templates will be included from other branches. `Microsoft.DotNet.Common.ProjectTemplates.6.0` from `release/6.0.1xx` branch will also be included to installer in `main` (to be 7.0.1xx).  

I made `Microsoft.DotNet.Common.ProjectTemplates.6.0` package in the PR: https://github.com/dotnet/templating/pull/4361

**Comment by MichaelSimons** [comment]
Let's take a step back and make sure all template versions are necessary for our source-build scenarios.  As I look at installer logic [here](https://github.com/dotnet/installer/blob/release/6.0.1xx/src/redist/targets/BundledTemplates.targets#L138), it looks like the 6.0 build only requires the 6.0 templates for source-build.  The other version references are needed when producing the MSI.  Source build doesn't support Windows.  @dsplaisted, @marcpopMSFT - can you confirm? Can we condition out the 5.0, 3.1, 3.0, and 2.1 template reference when building source-build or more abstractly non-win/non-arm?

**Comment by MichaelSimons** [comment]
This PR illustrates what I think we should consider to address the issue of source-build requiring the old templates - https://github.com/dotnet/installer/pull/13173.

**Comment by dsplaisted** [comment]
@MichaelSimons Yes, the previous template versions should not be needed for source build.  They are not included in the binary (ZIP) distribution on Windows, and I don't think they are included in the standalone installer either.  I think the reason that they are referenced and produced in the installer repo is because the older template versions are inserted into Visual Studio together with the .NET SDK.

Basically you should be good to ignore all of the templates from previous major versions in source build.
